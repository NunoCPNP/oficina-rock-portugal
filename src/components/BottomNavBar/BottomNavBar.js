import Link from 'next/link'
import useTranslation from 'next-translate/useTranslation'

import LanguageSelector from '../LanguageSelector'

import useAuth from '@/hooks/useAuth'

import { StyledNavigation, LinkContainer } from './BottomNavBar.styles'

const BottomNavBar = () => {
  const { t } = useTranslation('common')
  const { user } = useAuth()

  const hi = t(`hi`)

  return (
    <StyledNavigation>
      <div>{user && `${hi}, ${user.displayName} !`}</div>
      <LinkContainer>
        <ul>
          <li>
            <Link href="/bands">{t(`bands`)}</Link>
          </li>
          <li>
            <Link href="/men">{t(`men`)}</Link>
          </li>
          <li>
            <Link href="/women">{t(`women`)}</Link>
          </li>
          <li>
            <Link href="/acessories">{t(`acessories`)}</Link>
          </li>
          <li>
            <Link href="/promotions">{t(`promotions`)}</Link>
          </li>
          <li>
            <LanguageSelector />
          </li>
        </ul>
      </LinkContainer>
    </StyledNavigation>
  )
}

export default BottomNavBar
