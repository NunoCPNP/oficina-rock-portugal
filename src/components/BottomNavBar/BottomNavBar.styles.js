import mq from '@/styles/mediaQueries'
import styled from '@emotion/styled'

export const StyledNavigation = styled.div`
  display: grid;
  grid-template-columns: auto 1fr;
  padding: 1.4rem 4rem;
  background: var(--darkLiver);
  color: var(--snow);
  box-shadow: 0 0 10px 0 rgb(0 0 0 / 30%);

  ${mq('s')} {
    grid-template-columns: 1fr;
    grid-template-rows: auto auto;
    padding: ${({ user }) => (user ? '1rem' : 0)};
  }
`

export const UserName = styled.div`
  ${mq('s')} {
    justify-self: center;
    margin: ${({ user }) => (user ? '0.8rem' : '0')};
  }
`

export const LinkContainer = styled.div`
  display: flex;
  justify-content: flex-end;

  ul {
    display: inherit;
    margin: 0;
  }

  li {
    list-style: none;

    &:not(:last-child) {
      padding-right: 1.5rem;
    }

    a {
      text-decoration: none;
      color: inherit;

      &:hover {
        border-bottom: 1px solid var(--snow);
      }
    }
  }

  ${mq('s')} {
    justify-self: center;
    margin: 0.8rem;

    ul {
      padding-left: 0;
    }
  }
`
