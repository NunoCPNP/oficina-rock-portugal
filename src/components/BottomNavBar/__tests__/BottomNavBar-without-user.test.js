import { render } from '@testing-library/react'

import BottomNavBar from '../BottomNavBar'

jest.mock('@/hooks/useAuth', () => {
  return jest.fn(() => ({
    user: false,
  }))
})

describe('<BottomNavBar /> spec', () => {
  it('Should match snapshot', () => {
    const { container } = render(<BottomNavBar />)

    expect(container).toMatchSnapshot()
  })

  it('Should render translations keys', () => {
    const { getByText } = render(<BottomNavBar />)

    expect(getByText('common:men')).toBeInTheDocument()
    expect(getByText('common:women')).toBeInTheDocument()
  })

  it('Should render LanguageSelector Component', () => {
    const { queryAllByTestId } = render(<BottomNavBar />)

    expect(queryAllByTestId(/languageselector/i)).toHaveLength(1)
  })
})
