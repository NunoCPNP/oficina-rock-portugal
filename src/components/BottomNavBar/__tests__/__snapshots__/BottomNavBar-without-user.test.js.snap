// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomNavBar /> spec Should match snapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: auto 1fr;
  padding: 1rem 4rem;
  background: var(--darkLiver);
  color: var(--snow);
  box-shadow: 0 0 10px 0 rgb(0 0 0 / 30%);
}

@media only screen and (max-width: 480px) {
  .emotion-0 {
    grid-template-columns: 1fr;
    grid-template-rows: auto auto;
    padding: 0;
  }
}

@media only screen and (max-width: 480px) {
  .emotion-1 {
    justify-self: center;
    margin: 0;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-2 ul {
  display: inherit;
  margin: 0;
}

.emotion-2 li {
  list-style: none;
}

.emotion-2 li:not(:last-child) {
  padding-right: 1.5rem;
}

.emotion-2 li a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.emotion-2 li a:hover {
  border-bottom: 1px solid var(--snow);
}

@media only screen and (max-width: 480px) {
  .emotion-2 {
    justify-self: center;
    margin: 0.8rem;
  }

  .emotion-2 ul {
    padding-left: 0;
  }
}

.emotion-3 {
  padding-top: 0.2rem;
}

.emotion-3 a:hover {
  border-bottom: none!important;
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
  />
  <div
    class="emotion-2"
  >
    <ul>
      <li>
        <a
          href="/men"
        >
          common:men
        </a>
      </li>
      <li>
        <a
          href="/women"
        >
          common:women
        </a>
      </li>
      <li>
        <a
          href="/accessories"
        >
          common:accessories
        </a>
      </li>
      <li>
        <a
          href="/promotions"
        >
          common:promotions
        </a>
      </li>
      <li>
        <div
          class="emotion-3"
          data-testid="languageselector"
        />
      </li>
    </ul>
  </div>
</div>
`;
